design-documents/dto/proposal.md

```diff
@@ -180,6 +182,60 @@ class MyInjector implements InjectorProcessorInterface
 
 The attributes extension injector mechanism could also be used in mutable DTOs or existing classes.
 
+### Projection
+
+DTOs, and especially immutable DTOs, are widely used as read-only projections of existing data models or objects.
+
+An additional node `projection` can be specified in the `dto.xml` file containing the definition of the mapping procedure from an existing model.
+
+```
+<?xml version="1.0"?>
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="urn:magento:framework:Dto:etc/dto.xsd">
+
+    <interface
+            type="Test\DtoGenerator\Api\TestInterface"
+            mutable="false"
+    >
+        <property name="test1" type="string" nullable="true" />
+        <property name="test2" type="string" optional="true" />
+        <property name="test3" type="string" />
+        <property name="test_abc" type="string" />
+    </interface>
+
+    <class type="Test\DtoGenerator\Dto\Test" for="Test\DtoGenerator\Api\TestInterface" />
+
+    <projection for="Test\DtoGenerator\Api\TestInterface">
+        <from type="Magento\Catalog\Api\Data\ProductInterface">
+            <preprocessor type="Test\DtoGenerator\Processor\PreProcessor" />
+            <preprocessor type="Test\DtoGenerator\Processor\AnotherPreProcessor" />
+            <map from="sku" to="test1" />
```

Comment:
Will the return methods for projection attributes have the same return type as original attribute?